syntax="proto3";
package bank;


service Branch{
    rpc Withdraw(Event) returns (EventResponse) {}
    rpc Deposit(Event) returns (EventResponse) {}
    rpc Query(Event) returns (EventResponse) {}
    rpc Propagate_Deposit(Event) returns (ResponseStatus) {}
    rpc Propagate_Withdraw(Event) returns (ResponseStatus) {}
    rpc MsgDelivery(Event) returns (EventResponse) {}
    rpc InitStubs(CustNodes) returns (ResponseStatus){}
}


message CustomersI {
  int32 id =1;
  string type=2;
  repeated Event events = 3;
  int32 balance=4;
}

message Event {
  int32 id=1;
  string interface=2;
  int32 money=3;
}

message EventResponse {
  int32 id=1;
  string status=2;
  repeated InterfaceResponse respData = 3;
}

message InterfaceResponse{
  string interface = 1 ;
  string result = 2;
  int32 money = 3;
}

enum Result {
    FAILURE = 0;
    SUCCESS = 1;
}

message CustNodes {
  repeated NodePorts idports = 1;
}

message NodePorts {
  int32 id=1;
  int32 portNo=2;
}

message ResponseStatus {
  Result result =1;
}

message Balance {
  int32 balance = 1;
}

message EventList {
  repeated EventResponse event =1 ;
}

